x-pack/plugins/monitoring/server/telemetry_collection/get_es_stats.ts

```diff
@@ -107,5 +128,17 @@ export interface ESClusterStats {
 export function handleElasticsearchStats(response: estypes.SearchResponse<ESClusterStats>) {
   const clusters = response.hits?.hits || [];
 
-  return clusters.map((cluster) => cluster._source!);
+  return clusters.map((cluster) => {
+    const clusterStats = cluster._source!;
+
+    // tolerate metricbeat monitoring structure
+    if (clusterStats.elasticsearch) {
+      clusterStats.cluster_stats = clusterStats.elasticsearch.cluster.stats;
+      clusterStats.cluster_uuid = clusterStats.elasticsearch.cluster.id;
+      clusterStats.cluster_name = clusterStats.elasticsearch.cluster.name;
+      clusterStats.elasticsearch = undefined;
+    }
+
+    return clusterStats;
+  });
```

Comment:
@afharo, with this change, we tolerate the metricbeat monitoring structure and your [recent `cluster_uuid` fetch change](https://github.com/elastic/kibana/pull/182857) also properly works.
One caveat here, the `version` is a single in the structure but with metricbeat, there might be multiple versions under the nodes.

Reply:
Are you talking about scenarios where the same cluster runs nodes in different versions?

AFAIK, that should be temporary, right? Isn't the expectation that all nodes will eventually be in the same version? In that case, just reporting any version should be fine, IMO.

Reply:
Yes, as my understanding, metricbeat has a capability to fetch stats from multiple hosts (`elasticsearch-xpack.yml` module which has array of hosts config), and may have multiple versions.
I will at least put the first version resolved from its data.

Reply:
`version` is also available now. Thanks for you feedback.